src/workerd/jsg/filesystem.h

```diff
@@ -0,0 +1,244 @@
+#pragma once
+
+#include "jsg.h"
+
+namespace workerd::jsg::fs {
+
+// TODO(node-fs): Currently, all files and directories use a fixed last
+// modified time set to the Unix epoch. This is temporary.
+
+// Represents a file system node (file or directory) in the virtual file system.
+// This is an intentionally simplified alternative to the kj::FsNode interface
+// that minimizes it to only the types, methods, and metadata that are needed
+// for this jsg::VirtualFileSystem API.
+class Node: public kj::Refcounted {
+ public:
+  using Type = Node_Type;
+
+  Node() = default;
+  KJ_DISALLOW_COPY_AND_MOVE(Node);
+  virtual ~Node() noexcept(false) = default;
+
+  // Metadata about this filesystem node
+  struct Stat final {
```

Comment:
Nit: I prefer to avoid structs/classes in classes where possible.
Specifically here, you've already set the fs namespace.
I think stat even makes more sense as `fs::Stat` than `fs::Node::Stat`